plumbing/bootstrap.py

```diff
@@ -164,15 +162,24 @@ def collect_source_folders():
     data_folder = os.path.join(ROOT_DIR, 'data')
     valid_geocodes = load_geocodes()
 
-    logging.debug('Collecting sources in %s', data_folder)
+    logging.info('Collecting sources in %s', data_folder)
 
     for root, folders, _ in os.walk(data_folder):
         for folder in folders:
             _, name = os.path.split(folder)
             geocode = name.split('.')[0]
+
             if geocode in valid_geocodes:
-                logging.debug('Collected %s', folder)
+                logging.debug('%s: detected source folder', folder)
                 yield os.path.join(root, folder)
+                main_folder = os.path.join(root, folder)
+                yield main_folder
+
+                for _, sub_folders, _ in os.walk(main_folder):
```

Comment:
this is way too complicated... an os.walk inside an os.walk is surely not needed.